Monika After Story/game/zz_selector.rpy

```diff
@@ -3057,11 +3112,74 @@ style mas_selector_sidebar_vbar:
 #   cancel - label to jump to when canceling
 #   restore - label to jump to when restoring
 #   remover - remover display item, if appropriate. Can be None
-screen mas_selector_sidebar(items, mailbox, confirm, cancel, restore, remover=None):
+#   filter_map - the list of selectables for each category shown in the menu. Can be None
+screen mas_selector_sidebar(items, mailbox, confirm, cancel, restore, remover=None, filter_map=None):
     zorder 50
-#    modal True
 
     $ sel_frame_vsize = mailbox.read_frame_vsize()
+            
+    # only add menu if filter_map is provided and has more than one option
+    if filter_map and len(filter_map) > 1:
+        #filter dropdown button
+        frame:   
+            area (960, 3, 50, 40)
+            background None           
+
+            button:
+                if show_filter: 
+                    style "filter_dropdown_down"
+                else:
+                    style "filter_dropdown_up"
+                action [If(show_filter == False, true=SetVariable('show_filter', True), false=SetVariable('show_filter', False)),Jump("mas_selector_sidebar_select_midloop")]  
```

Comment:
midloop jumping is mainly for showing the dialogue box. The way screens work is that they refresh continuously, so setting the variable is enough to make the screen change next refresh.

Reply:
Fixed in new commit.